runtime/src/iree/hal/drivers/vulkan/base_buffer.c

```diff
@@ -36,6 +36,74 @@ static bool iree_hal_vulkan_is_memory_type_usable(VkMemoryPropertyFlags flags) {
          !iree_all_bits_set(flags, VK_MEMORY_PROPERTY_PROTECTED_BIT);
 }
 
+iree_status_t iree_hal_vulkan_find_memory_type(
+    const VkPhysicalDeviceProperties* device_props,
+    const VkPhysicalDeviceMemoryProperties* memory_props,
+    const iree_hal_buffer_params_t* IREE_RESTRICT params,
+    uint32_t* out_memory_type_index) {
+  *out_memory_type_index = 0;
+
+  VkMemoryPropertyFlags require_flags = 0;
+  VkMemoryPropertyFlags prefer_flags = 0;
+  if (iree_all_bits_set(params->type, IREE_HAL_MEMORY_TYPE_DEVICE_LOCAL)) {
+    if (iree_all_bits_set(params->type, IREE_HAL_MEMORY_TYPE_HOST_VISIBLE)) {
+      // Device-local, host-visible.
+      require_flags |= VK_MEMORY_PROPERTY_HOST_VISIBLE_BIT;
```

Comment:
Super nit: order it to follow DEVICE_LOCAL and then HOST_VISIBLE is more consistent with the above checks.

Reply:
it's ordered to match the local require_flags/prefer_flags order - being device local and host visible requires host visible but only prefers device local (if there's no device-local + host-visible memory then we still need to meet the host-visible requirement and then it's just slow)